net/mptcp/mptcp_ctrl.c

```diff
@@ -92,6 +92,22 @@ static int proc_mptcp_path_manager(struct ctl_table *ctl, int write,
 	return ret;
 }
 
+static int proc_mptcp_available_path_manager(ctl_table *ctl, int write,
+				   void __user *buffer, size_t *lenp,
+				   loff_t *ppos)
+{
+	char val[MPTCP_PM_NAME_MAX];
```

Comment:
val is too small here, as maxlen is set to MPTCP_PM_BUF_MAX. You will overflow buffers here.